src/backend/app.js

```diff
@@ -1,30 +1,36 @@
 const express = require('express');
 const app = express();
-// const fs = require("fs");
+const fs = require("fs");
 const tvShows = require("./ShowsList");
 const path = require('path');
 const imageRes = require('./image/ResolveID');
 const port = process.env.PORT || 3001;
 
+//downloading movie data
 app.get('/rest/show', (req, res)=>{
-    
+   console.log('rest api requested'); 
     res.type('json');
     res.end(JSON.stringify(tvShows.tvShows));
 });
 
-// app.get('/image/:id', (req, res)=> {
-//     console.log(`the param is ${req.params.name}`);
-// });
+//downloading the images
+//app.use(express.static(path.join(__dirname, './image/covers')));
+app.get('/images/:id', function (req, res) {
+    let imageExists = tvShows.tvShows.find((tvShow) => tvShow.id === req.params.id);
 
-app.use(express.static(path.join(__dirname, './image/covers')));
-app.get('/images/:name', function (req, res) {
-    res.sendFile(path.join(__dirname,  './image/covers', `${req.params.name}.jpg`));
+    if(!imageExists){
+        res.redirect('/not-found');
```

Comment:
So if an image is broken, you strictly redirect to the `not-found` page. I personally find this strategy a bit aggressive with the user!